src/text_format.cc

```diff
@@ -30,6 +30,7 @@ bool ParseTextMessage(const std::string& data, protobuf::Message* output) {
   TextFormat::Parser parser;
   parser.SetRecursionLimit(100);
   parser.AllowPartialMessage(true);
+  parser.AllowUnknownField(true);
```

Comment:
what about binary_format?

Reply:
Binary format is the serialized data, right? Protos should already allow unknown fields when deserializing (that is a well-known property of protos). Just not for text format.